### PR TITLE
Scaffold db

### LINKED ISSUE #19: Scaffold the database

### PR DESCRIPTION
**Pull request by calebschoepp**
## Changes
- Write api README
- Write db README
- Guide for running Postgres locally
- Update tech stack guide

## TODO
- Write 1st migration i.e. DB schema. Moving this to another PR to keep scope smaller

Closes #19

### CODE REVIEW

File: api/db/migrations/001_initial_schema.up.sql [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+CREATE TRIGGER update_time_resumes BEFORE
+UPDATE
+  ON resumes FOR EACH ROW EXECUTE PROCEDURE update_modified_column();
+
+CREATE TRIGGER update_time_reviews BEFORE
+UPDATE
+  ON reviews FOR EACH ROW EXECUTE PROCEDURE update_modified_column();
+
+-- Table for each resume reviewer, interviewer
+-- Table for feedback?
```
Comment (cyrusdiego) [inline api/db/migrations/001_initial_schema.up.sql:61]: nit: newline

File: api/db/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Starting Postgres
+
+`docker run --rm --name pg-docker -e POSTGRES_PASSWORD=postgres -e POSTGRES_USER=postgres -e POSTGRES_DB=compe-plus -d -p 5432:5432 -v $HOME/docker/volumes/postgres:/var/lib/postgresql/data postgres:13`
+
+Verify that it is running with `docker ps` or kill it with `docker kill pg-docker`
+
+## Connecting to Postgres
+
+`psql -h localhost -U postgres -d compe-plus` (Password is "postgres")
```
Comment (armiantos) [inline api/db/README.md:24]: Does this imply we need to have postgres installed in our machines or should this run under something like `docker exec ...` to the container we ran in the previous step?

**Review by armiantos — COMMENTED** [review]
The READMEs look great! Just had one comment and do you want to create schema v1 in this PR or is that going to be in a separate one?

File: api/README.md [hunk 3]
```diff
@@ -1,3 +1,61 @@
...
+
+# Local Development
+
+To start a local development server run `npm run start:dev`. This will automatically pull in new changes as you update files.
+
+To test the code run `npm test`. You can also run `npm run test:watch` for an interactive test session.
+
+We also have a few scripts to keep the code quality up to snuff. Run `npm run format` to format all the code with Prettier and run `npm run lint` to make sure that ESLint doesn't complain about anything.
+
+The API makes use of a number of services e.g., database, blog storage, email. See the [dependencies](#Dependencies) section for more on how to locally emulate these services.
```
Comment (cyrusdiego) [inline api/README.md:17]: idk if it's just the Github markdown renderer but the link isn't working but I tried doing it locally and it worked fine in vscode. i'm just gonna assume this link properly.

File: api/db/migrations/001_initial_schema.down.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+-- TODO
```
Comment (cyrusdiego) [inline api/db/migrations/001_initial_schema.down.sql:1]: what's the difference between `.down.sql` and `.up.sql`

**Review by cyrusdiego — COMMENTED** [review]
Just had some questions, other than that I think it looks good. Probably let @armiantos approve after you guys resolve his comment.

File: api/db/migrations/001_initial_schema.down.sql (hunk 4 above)
Comment (armiantos) [reply api/db/migrations/001_initial_schema.down.sql:1]: I'm assuming how to rollback and how to update.

File: api/README.md (hunk 3 above)
Comment (armiantos) [reply api/README.md:17]: The anchor link? It just scrolls to the dependencies section, it works for me when I open this branch on a separate page.

File: api/db/README.md (hunk 2 above)
Comment (calebschoepp) [reply api/db/README.md:24]: I callout installing `psql` earlier in the README with `sudo apt-get install postgresql-client`. This installs `psql` and not all of postgres.

**Comment by calebschoepp** [comment]
> The READMEs look great! Just had one comment and do you want to create schema v1 in this PR or is that going to be in a separate one?

Leaving schema for next PR

File: api/README.md (hunk 3 above)
Comment (calebschoepp) [reply api/README.md:17]: Yeah works locally for me too

Comment (cyrusdiego) [reply api/README.md:17]: ok ignore me haha

File: api/db/migrations/001_initial_schema.down.sql (hunk 4 above)
Comment (calebschoepp) [reply api/db/migrations/001_initial_schema.down.sql:1]: Yep. Normally you only need to run up migrations but you need down ones to revert if you screwed up. I can teach you guys more about migrations in a meeting if you want.

File: api/db/README.md (hunk 2 above)
Comment (armiantos) [reply api/db/README.md:24]: Ah my bad I missed that. I'm assuming for Mac and Windows they'll have to get a postgresql client container?

**Review by armiantos — APPROVED** [review]
With the actual schema being developed in another PR, the READMEs and tech clarifications are👌

**Comment by calebschoepp** [comment]
> Ah my bad I missed that. I'm assuming for Mac and Windows they'll have to get a postgresql client container?

They could do it via Docker probably like you're saying. Also though I'm sure they could just directly install the client somehow.

I basically just wrote all the instructions assuming Linux. We should do a quick survey of the operating systems used by the 5 of us and update the docs appropriately in the future.

**Comment by armiantos** [comment]
Yep, sounds good to me :+1:

File: api/db/migrations/001_initial_schema.down.sql (hunk 4 above)
Comment (cyrusdiego) [reply api/db/migrations/001_initial_schema.down.sql:1]: yes please haha, I wanted to learn more backend stuff

**Pull request merged**